### PR TITLE
fix(analytics): adding traits for analytics

### PR DESCRIPTION
**Pull request by pedroapfilho**

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/analytics.ts [hunk 1]
```diff
@@ -1,47 +1,54 @@
...
 import { AnalyticsKey } from 'middleware/analyticsMiddleware/types'
 import { generateUniqueUserId } from 'middleware/analyticsMiddleware/utils'
-import { v4 } from 'uuid'
 
-const queueCallback = async (rawEvents) => {
+const queueCallback = async (
+  rawEvents: {
+    key: AnalyticsKey
+    payload: AnalyticsValue
+  }[]
```
Comment (milan-bc) [inline packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/analytics.ts:10]: maybe to export this to be type and the use it here like
```
type RawEventsType = {
    key: AnalyticsKey
    payload: AnalyticsValue
  }
```

File: packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/analytics.ts [hunk 2]
```diff
@@ -1,47 +1,54 @@
...
 
   const events = rawEvents.map((event) => {
     const name = event.key
 
-    const { analyticsType, id, nabuId, originalTimestamp, ...properties } = event.payload
+    const { id, originalTimestamp, ...properties } = event.payload.properties
 
     return {
-      id: generateUniqueUserId(id || randomId),
+      id: id ? generateUniqueUserId(id) : null,
```
Comment (milan-bc) [inline packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/analytics.ts:39]: is it ok if we pass `null` here?

**Review by milan-bc — COMMENTED** [review]
I left some minor comments, beside of that looks ok to me

File: packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/analytics.ts (hunk 2 above)
Comment (pedroapfilho) [reply packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/analytics.ts:39]: yep

File: packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/analytics.ts (hunk 1 above)
Comment (pedroapfilho) [reply packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/analytics.ts:10]: Good one!

**Review by milan-bc — APPROVED** [review]

**Pull request merged**